src/program_generator/pg_marginal.py

```diff
@@ -96,7 +86,7 @@ def _generate_marginal(self, opcode, max_op_count, shuffle_counts):
     else:
       pass
 
-    op_counts = list(range(0, max_op_count))
+    op_counts = list(range(0, max_op_count + 1, step_op_count))
```

Comment:
unrelated to the main change in this PR - we expect `maxOpCount` to be inclusive, so we need to `+1`